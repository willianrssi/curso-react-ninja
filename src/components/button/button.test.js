'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Button from './index'

it('Should button success to match snapshot', () => {
  const tree = renderer.create(
    <Button onClick={() => null} kind='success'>
      Click me success
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should button danger to match snapshot', () => {
  const tree = renderer.create(
    <Button onClick={() => null} kind='danger'>
      Click me success
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should button default to match snapshot', () => {
  const tree = renderer.create(
    <Button onClick={() => null} >
      Click me success
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Button with props kind="danger" should has class "-danger" ', () => {
  const wrapper = shallow(
    <Button onClick={() => null} kind='danger' >
      Click me success
    </Button>
  )
  expect(wrapper.hasClass('-danger')).toBe(true)
  expect(wrapper.hasClass('-success')).toBe(false)
})

it('Button with props kind="success" should has class "-success" ', () => {
  const wrapper = shallow(
    <Button onClick={() => null} kind='success' >
      Click me success
    </Button>
  )
  expect(wrapper.hasClass('-success')).toBe(true)
  expect(wrapper.hasClass('-danger')).toBe(false)
})
