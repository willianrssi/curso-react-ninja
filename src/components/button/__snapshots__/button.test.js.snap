// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should button danger to match snapshot 1`] = `
<button
  className="button -danger"
  onClick={[Function]}
>
  Click me success
</button>
`;

exports[`Should button default to match snapshot 1`] = `
<button
  className="button "
  onClick={[Function]}
>
  Click me success
</button>
`;

exports[`Should button success to match snapshot 1`] = `
<button
  className="button -success"
  onClick={[Function]}
>
  Click me success
</button>
`;
